rustfst/src/algorithms/lazy/cache/simple_hash_map_cache.rs

```diff
@@ -91,6 +95,8 @@ impl<W: Semiring> FstCache<W> for SimpleHashMapCache<W> {
                 noepsilons += 1;
             }
         }
+        self.num_known_trs
+            .fetch_add(trs.trs().len(), Ordering::SeqCst);
```

Comment:
Indeed there is an issue here if the insert_trs is called twice with a different trs lenth 😬 
Do you think you could handle this case ?

Reply:
Following your comments, computing this value on the fly is the most reliable way.